### PR TITLE
refactor: Move flyPad to fbw-common

### PR DESCRIPTION
**Pull request by frankkopp**
## Summary of Changes
Moving the flyPad and associated shared modules to the fbw-common folder.

Discord username (if different from GitHub): cdr_maverick

## Testing instructions
Regression test on the A320 and A380 for the complete flyPad (incl. translations, integrations, etc.)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/localization/README.md [hunk 1]
```diff
@@ -6,6 +6,16 @@ TOC:
 - [Build Process](#build-process)
 - [Updating the Source File](#update-source-file)
 
+OBS:
+
+<span style="color: red">
+fylPad Localization files are now in the fbw-common EFB section
```
Comment (2hwk) [inline fbw-a32nx/src/localization/README.md:12]:
```suggestion
flyPad Localization files are now in the fbw-common EFB section
```

File: fbw-a380x/src/systems/failures/src/a380.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// One can rightfully argue that this constant shouldn't be located in @flybywiresim/failures.
```
Comment (2hwk) [inline fbw-a380x/src/systems/failures/src/a380.ts:1]:
```suggestion
// SPDX-License-Identifier: GPL-3.0
...
// One can rightfully argue that this constant shouldn't be located in @flybywiresim/failures.
```

Add GPL Header

File: fbw-common/src/systems/datalink/common/src/types/AtsuFlightPhase.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export enum AtsuFlightPhase {
```
Comment (2hwk) [inline fbw-common/src/systems/datalink/common/src/types/AtsuFlightPhase.ts:1]:
```suggestion
// Copyright (c) 2023-2024 FlyByWire Simulations
// SPDX-License-Identifier: GPL-3.0

export enum AtsuFlightPhase {
```

File: fbw-common/src/systems/shared/src/failures/index.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export { FailuresConsumer } from './failures-consumer';
```
Comment (2hwk) [inline fbw-common/src/systems/shared/src/failures/index.ts:1]:
```suggestion
// Copyright (c) 2023-2024 FlyByWire Simulations
// SPDX-License-Identifier: GPL-3.0

export { FailuresConsumer } from './failures-consumer';
```

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A380/EFB/mapConfig.json [hunk 5]
```diff
@@ -0,0 +1,118 @@
+﻿{
```
Comment (2hwk) [inline fbw-a380x/src/base/flybywire-aircraft-a380-842/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A380/EFB/mapConfig.json:1]: is mapConfig.json a new configuration? Document this in the PR if so

Comment (2hwk) [inline fbw-a380x/src/base/flybywire-aircraft-a380-842/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A380/EFB/mapConfig.json:1]: is mapConfig.json a new configuration? Document this in the PR if so

**Review by 2hwk — CHANGES_REQUESTED** [review]

File: fbw-a380x/src/base/flybywire-aircraft-a380-842/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A380/EFB/mapConfig.json (hunk 5 above)
Comment (frankkopp) [reply fbw-a380x/src/base/flybywire-aircraft-a380-842/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A380/EFB/mapConfig.json:1]: it is a copy of the A32NX - no changes but the A380X should have its own copy in its base files

**Comment by Popespice** [comment]
Quality Assurance Report

Discord          : Popespice
Object of testing: #8473 
Tier of Testing  : 1
Date             : 02212024

Testing Process: 
1. Log into Sim
2. Power on aircraft as normal
3. Turn on Flypad
4. Press every interactable button on flypad (failures, settings, presets, all of it)
5. Observe action works as intended

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
Works as expected

Media:       
N/A

**Comment by Popespice** [comment]
Please note, the QA test as reported above is initial findings only on A32NX.

**Review by 2hwk — APPROVED** [review]

**Comment by tylerclt** [comment]
Quality Assurance Trainee Report

Discord: Tylerb780#3594
Object of testing: #8473 
Tier of Testing: 1
Date: 23/02/2024

Testing Process: 
1. Launched Sim 
2. Loaded up the A32NX
3. Once Loaded in, powered aircraft on
4. Turned the FlyPad on and pressed every button that was interactable
5. Verified that the buttons and their intended actions worked.

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
Functioned as expected!

**Pull request merged**